### PR TITLE
[tune] Retry cloud sync up/down/delete on fail

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Syncing of checkpoints to/from cloud storage or deleting them sometimes fails due to endpoint issues (i.e. AWS/GS communication). In these cases we usually want to retry the command as failures are often stochastic. This PR hopefully improves test rate of our durable upload release tests.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/sync_client.py [hunk 1]
```diff
@@ -251,6 +256,7 @@ def __init__(
         self.logfile = None
         self._closed = False
         self.cmd_process = None
+        self._last_cmd = None
```
Comment (xwjiang2010) [inline python/ray/tune/sync_client.py:259]: Could we add a comment here what this is for?

**Review by xwjiang2010 — APPROVED** [review]
Looks great. Just a nit.

**Comment by krfricke** [comment]
Test run: https://buildkite.com/ray-project/periodic-ci/builds/2617

**Comment by xwjiang2010** [comment]
Test run fails :(

**Pull request merged**